internal/api/impl/v1/dashboard/schemas/validator.go

```diff
@@ -30,39 +30,70 @@ import (
 // Validator can be used to run checks on panels, based on cuelang definitions
 type Validator interface {
 	Validate(panels map[string]json.RawMessage) error
-	LoadSchemas()
+	LoadCharts()
+	LoadQueries()
 }
 
 type validator struct {
-	schemasConf config.Schemas
-	context     *cue.Context
-	baseDef     cue.Value
-	schemas     *sync.Map
+	schemasConf    config.Schemas
+	context        *cue.Context
+	baseChartDef   cue.Value
+	chartsSchemas  *sync.Map
+	baseQueryDef   cue.Value
+	queriesSchemas *sync.Map
```

Comment:
I have the feeling it will be interesting to have a dedicated structure that will hold the load of the schemas.
My thought is based on the fact you have for the queries and the charts 2 similars attributes.

If you have another struct containing 2 attributes `baseDef` and `schemas` then in the struct `validator` you just have one attribute per kind of plugin we are supporting. So here `charts` and `queries`.

Also for later, probably it will help to move the variables to cuelang. A move that likely will happen.